### ISSUE TITLE
Support writing linting issues to file/etc

### ISSUE THREAD

**Issue by onwsk8r** [golangci/golangci-lint#2058]
**Is your feature request related to a problem? Please describe.**

First, forcing `printers/*` output to `os.Stdout` and circumventing `logrus` is probably more of a bug than a feature (the printers use [this pattern](https://github.com/golangci/golangci-lint/blob/master/pkg/printers/text.go#L64) to write [here](https://github.com/golangci/golangci-lint/blob/master/pkg/logutils/out.go#L8), which is [unnecessary](https://github.com/fatih/color/blob/master/color.go#L27) [complexity](https://github.com/mattn/go-colorable/blob/master/colorable_others.go#L23)). Writing errors to stdout is certainly unexpected, will make life interesting for future logrus changes, and relies on `/dev/stdout` being what you think it is. You might find exceptions in corner cases like, say, CI environments.

To the point: generating reports (read: from `artifacts`) is a common, often necessary, but currently unsupported practice to:

- prevent developers, reviewers, and anyone else involved from having to examine the output of every single job in the pipeline
- export data for reanalysis
- generate supporting documentation for releases (this is especially important in the compliance world)

**Describe the solution you'd like**

The application should have a CLI flag and configuration file option to send the output of `printers/*` to a file

<details>

Since everything uses `fmt.Fprint()` and not `_, err := fmt.Fprint()` but lacks a `// nolint`, I'm guessing this code isn't from last week :) The simplest solution would be to add a flag (`--out-file`?) and set the `io.Writer` for `fmt.Fprint` accordingly. That would take like five minutes, resulting in:

```shell
$ golangci-lint run -v --out-format code-climate --out-file "reports/oops.json"
INFO [config_reader] Config search paths: [./ /foo /]
INFO [config_reader] Used config file .golangci.yml
# ...
INFO [runner] linters took 12.129263493s with stages: goanalysis_metalinter: 12.124616398s
INFO File cache stats: 15 entries of total size 22.2KiB
$ cat reports/oops.json
[{"description":"gomnd: mnd: Magic number: 250, in \u003cassign\u003e detected","fingerprint":"195260A29896B51E8E882F52DCD65243","location":{"path":"some/file.go","lines":{"begin":13}}}]
```

The caveat here is that errors are not printed to the terminal, which is desirable.

A more robust solution would be to hook the printers into a child logrus with an optional different writer or hook or something, but this is backwards-incompatible with all the people hacking things together as below unless you also use an empty formatter.

</details>

**Describe alternatives you've considered**

- `>/some/file` explodes if there is any other output (such as `WARN [runner] The linter 'scopelint' is deprecated`).
- `| grep -v '^(INFO|WARN|ERR|FATA) >/some/file` excludes results from my `WARN` package and is horrid
- `| tee /some/file` and manually grok the output like some kind of cave man

**Additional context**

I use `-v` to get memory usage, warnings about deprecated linters, and probably other things. From an operations perspective, that's really handy, and you should get some kind of award for building that in.

In GH and Travis, which are the most widely-used solutions among the anti-money-spending crowd, it's not that important. For CloudBees, CircleCI, anything on-prem (like Actions self-hosted runners), Gitlab which has [built-in support](https://docs.gitlab.com/ee/ci/yaml/#artifactsreportscodequality), etc, reports are the preferred, um, reporter. Some people don't want those issues being as accessible as, say, container logs.

**Comment by ldez** [comment]
Duplicate of #481

**Comment by ldez** [comment]
FYI:

```console
$ golangci-lint run --out-format=json > foo.json
WARN [runner] The linter 'golint' is deprecated (since v1.41.0) due to: The repository of the linter has been archived by the owner.  Replaced by revive. 
$ cat foo.json 
{"Issues":[{"FromLinter":"gosec","Text":"G204: Subprocess launched with variable","Severity":"","SourceLines":["\t\t\tcmd := exec.CommandContext(ctx, name, args...)"],"Replacement":null,"Pos":{"Filename":"pkg/topology/store/store.go","Offset":0,"Line":40,"Column":11},"ExpectNoLint":false,"ExpectedNoLintLinter":""}],"Report":{"Warnings":[{"Tag":"runner","Text":"The linter 'golint' is deprecated (since v1.41.0) due to: The repository of the linter has been archived by the owner.  Replaced by revive."}],"Linters":[{"Name":"govet","Enabled":true,"EnabledByDefault":true},{"Name":"bodyclose"},{"Name":"noctx","Enabled":true},{"Name":"errcheck","Enabled":true,"EnabledByDefault":true},{"Name":"golint","Enabled":true},{"Name":"rowserrcheck"},{"Name":"staticcheck","Enabled":true,"EnabledByDefault":true},{"Name":"unused","Enabled":true,"EnabledByDefault":true},{"Name":"gosimple","Enabled":true,"EnabledByDefault":true},{"Name":"stylecheck","Enabled":true},{"Name":"gosec","Enabled":true},{"Name":"structcheck","Enabled":true,"EnabledByDefault":true},{"Name":"varcheck","Enabled":true,"EnabledByDefault":true},{"Name":"interfacer"},{"Name":"unconvert","Enabled":true},{"Name":"ineffassign","Enabled":true,"EnabledByDefault":true},{"Name":"dupl"},{"Name":"goconst","Enabled":true},{"Name":"deadcode","Enabled":true,"EnabledByDefault":true},{"Name":"gocyclo","Enabled":true},{"Name":"cyclop"},{"Name":"gocognit","Enabled":true},{"Name":"typecheck","Enabled":true,"EnabledByDefault":true},{"Name":"asciicheck","Enabled":true},{"Name":"gofmt","Enabled":true},{"Name":"gofumpt","Enabled":true},{"Name":"goimports","Enabled":true},{"Name":"goheader","Enabled":true},{"Name":"gci","Enabled":true},{"Name":"maligned"},{"Name":"depguard","Enabled":true},{"Name":"misspell","Enabled":true},{"Name":"lll"},{"Name":"unparam","Enabled":true},{"Name":"dogsled","Enabled":true},{"Name":"nakedret","Enabled":true},{"Name":"prealloc"},{"Name":"scopelint"},{"Name":"gocritic","Enabled":true},{"Name":"gochecknoinits","Enabled":true},{"Name":"gochecknoglobals","Enabled":true},{"Name":"godox","Enabled":true},{"Name":"funlen","Enabled":true},{"Name":"whitespace","Enabled":true},{"Name":"wsl"},{"Name":"goprintffuncname","Enabled":true},{"Name":"gomnd"},{"Name":"goerr113"},{"Name":"gomodguard","Enabled":true},{"Name":"godot","Enabled":true},{"Name":"testpackage"},{"Name":"nestif","Enabled":true},{"Name":"exportloopref","Enabled":true},{"Name":"exhaustive"},{"Name":"sqlclosecheck"},{"Name":"nlreturn"},{"Name":"wrapcheck"},{"Name":"thelper","Enabled":true},{"Name":"tparallel"},{"Name":"exhaustivestruct"},{"Name":"errorlint","Enabled":true},{"Name":"paralleltest"},{"Name":"makezero","Enabled":true},{"Name":"forbidigo","Enabled":true},{"Name":"ifshort"},{"Name":"predeclared","Enabled":true},{"Name":"revive","Enabled":true},{"Name":"durationcheck","Enabled":true},{"Name":"wastedassign","Enabled":true},{"Name":"importas","Enabled":true},{"Name":"nilerr","Enabled":true},{"Name":"forcetypeassert"},{"Name":"gomoddirectives","Enabled":true},{"Name":"promlinter","Enabled":true},{"Name":"tagliatelle","Enabled":true},{"Name":"nolintlint","Enabled":true}]}}
```

It works because the logs (`WARN ...`) are printed on `os.Stderror` and the report on `os.Stdout`
